power/src/power.rs

```diff
@@ -109,6 +230,44 @@ impl TestKind {
                 n - (*rho as f64) - 1.0,
                 es.powi(2) * n,
             )),
+            TestKind::ANCOVA { k, q, p } => Box::new(NoncentralF::new(
+                *q as f64,
+                n - *k as f64 - *p as f64 - 1.0,
+                es.powi(2) * n,
+            )),
+            TestKind::OneWayANOVA { k } => Box::new(NoncentralF::new(
+                *k as f64 - 1.0,
+                n - *k as f64,
+                es.powi(2) * n,
+            )),
+            TestKind::TwoWayANOVA { k, q } => {
+                Box::new(NoncentralF::new(*q as f64, n - *k as f64, es.powi(2) * n))
+            }
+
+            TestKind::BetweenRepeatedANOVA { k, m, rho } => {
+                let u = *m as f64 / (1.0 + ((*m as f64 - 1.0) * *rho));
+                Box::new(NoncentralF::new(
+                    *k as f64 - 1.0,
+                    n - *k as f64,
+                    es.powi(2) * u * n,
+                ))
+            }
+            TestKind::WithinRepeatedANOVA { k, m, rho, epsilon } => {
+                let u = *m as f64 / (1.0 - *rho);
+                Box::new(NoncentralF::new(
+                    (*m as f64 - 1.0) * *epsilon,
+                    (n - *k as f64) * (*m as f64 - 1.0) * *epsilon,
+                    es.powi(2) * u * n * *epsilon, // NOTE: G*Power paper is missing the epsilon.
```

Comment:
```suggestion
                    es.powi(2) * u * n * *epsilon, // G*Power paper is missing the epsilon.
```
Or comment on line above? We're going well over the normal line length limit, right?